cmd/installer/main.go

```diff
@@ -17,66 +19,105 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-func parseOptions(uninstall *bool, options *install.SafeOptions) *rest.Config {
-	var etcdCA, etcdCert, etcdKey, tlsCA, tlsCert, tlsKey string
-	var logLevel string
-	var pullPolicy string
+type certBundleSource struct {
+	ca, cert, key string
+}
+
+func (s certBundleSource) load() (*install.TLSBundle, error) {
+	if (s.ca != "" || s.cert != "" || s.key != "") &&
+		(s.ca == "" || s.cert == "" || s.key == "") {
+		return nil, errors.New("ca-file, cert-file and key-file must be set altogether or not at all")
+	}
+	var caBytes, certBytes, keyBytes []byte
+	loadFile(s.ca, &caBytes)
+	loadFile(s.cert, &certBytes)
+	loadFile(s.key, &keyBytes)
+	if caBytes == nil {
+		return nil, nil
+	}
+	bundle, err := install.NewTLSBundle(caBytes, certBytes, keyBytes)
+	if err != nil {
+		return nil, err
+	}
+	return bundle, nil
+}
+
+type additionalFlags struct {
+	etcdBundle certBundleSource
+	tlsBundle  certBundleSource
+	logLevel   string
+	pullPolicy string
+	kubeconfig string
+	uninstall  bool
+}
+
+func parseFlags(customFlags *additionalFlags, options *install.SafeOptions) {
 	flag.BoolVar(&options.Network.ShouldUseHost, "network-host", false, "Use network host")
 	flag.StringVar(&options.OptionsCommon.Namespace, "namespace", "docker", "Namespace in which to deploy")
 	flag.DurationVar(&options.OptionsCommon.ReconciliationInterval, "reconciliation-interval", constants.DefaultFullSyncInterval, "Interval of reconciliation loop")
 	flag.StringVar(&options.OptionsCommon.Tag, "tag", "latest", "Image tag")
-	flag.StringVar(&pullPolicy, "pull-policy", string(corev1types.PullAlways), fmt.Sprintf("Image pull policy (%q|%q|%q)", corev1types.PullAlways, corev1types.PullNever, corev1types.PullIfNotPresent))
+	flag.StringVar(&customFlags.pullPolicy, "pull-policy", string(corev1types.PullAlways),
+		fmt.Sprintf("Image pull policy (%q|%q|%q)", corev1types.PullAlways, corev1types.PullNever, corev1types.PullIfNotPresent))
```

Comment:
nit: this file need some `fmt` love 😄

Reply:
Well, the linter complained about line length... Had to split this one :/